src/nmsgtool.c

```diff
@@ -411,6 +448,47 @@ setup_nmsg_input(nmsgtool_ctx *c, nmsg_input_t input) {
 
 /* Private functions. */
 
+#ifdef HAVE_PROMETHEUS
+
+static void
+init_prometheus_counters(void)
+{
+	const char *label = "nmsgtool";
+
+	/* NMSG payload counters */
+	INIT_PROM_CTR_L(total_payloads_in, "total_payloads_in", "total number of nmsg payloads received", label);
+	assert(total_payloads_in != NULL);
+	INIT_PROM_CTR_L(total_payloads_out, "total_payloads_out", "total number of nmsg payloads sent", label);
+	assert(total_payloads_out != NULL);
+
+	/* NMSG container counters */
+	INIT_PROM_CTR_L(total_container_recvs, "total_container_recvs", "total number of nmsg containers received", label);
+	assert(total_container_recvs != NULL);
+	INIT_PROM_CTR_L(total_container_drops, "total_container_drops", "total number of nmsg containers lost", label);
+	assert(total_container_drops != NULL);
+}
+
+static int nmsgtool_prom_handler(void *clos) {
```

Comment:
Add comment with: 
/* This is the prometheus callback function.  clos is a nmsg_io_t, which gives us the handle to get nmsg statistics.  Always returns 0, which means success. */

Alternatively for that last sentence: check for nmsg_io_get_stats() returning nmsg_res_failure and if so, return a -1 fail code from this.